code/game/objects/items/devices/scanners.dm

```diff
@@ -120,7 +121,7 @@ GENE SCANNER
 						"<span class='notice'>You analyze [M]'s vitals.</span>")
 
 	if(scanmode == 0)
-		healthscan(user, M, advanced)
+		healthscan(user, M, mode, advanced)
```

Comment:
Can't you just remove the var/mod and replace this line with
```suggestion
		healthscan(user, M, advanced=advanced)
```

Reply:
yeah that works